OpenSim/Simulation/Model/Station.h

```diff
@@ -37,38 +36,56 @@ class Body;
- * to origin of a Body), a FixedFrame (affixed to another PhysicalFrame) or any 
- * other PhysicalFrame. The main functionality provided by Station is to find its 
- * location in any Frame.
+ * A Station is a Point fixed to and located on a PhysicalFrame, which can be a 
+ * Body (w.r.t. the Body origin), Ground or any PhysicalOffsetFrame.
+ * A Station is an analogous physical Point to a PhyscialFrame where joints,
+ * constraints and forces can be attached and/or applied.
  *
- * @author Ayman Habib
- * @version 1.0
+ * @author Ayman Habib, Ajay Seth
  */
-class OSIMSIMULATION_API Station : public ModelComponent {
-OpenSim_DECLARE_CONCRETE_OBJECT(Station, ModelComponent);
+class OSIMSIMULATION_API Station : public Point {
+OpenSim_DECLARE_CONCRETE_OBJECT(Station, Point);
 public:
     //==============================================================================
     // PROPERTIES
     //==============================================================================
+    /** @name Property declarations
+    These are the serializable properties associated with a Station. **/
+    /**@{**/
     OpenSim_DECLARE_PROPERTY(location, SimTK::Vec3,
-        "The location (Vec3) of the station in a reference frame. "
-        "Frame is specified as Connector.");
+        "The location of the station in its reference frame.");
+    /**@}**/
 
 public:
     //--------------------------------------------------------------------------
     // CONSTRUCTION
     //--------------------------------------------------------------------------
+    /** Default constructor */
     Station();
+
+    /** Convenience constructor
+    @param[in] frame     PhysicalFrame in which the Station is located.
+    @param[in] location  Vec3 location of the station in its PhysicalFrame */
+    Station(const PhysicalFrame& frame, const SimTK::Vec3& location);
+
     virtual ~Station();
     /** getter of Reference Frame off which the Station is defined */
-    const OpenSim::PhysicalFrame& getReferenceFrame() const;
+    const PhysicalFrame& getReferenceFrame() const;
     /** setter of Reference Frame off which the Station is defined */
     void setReferenceFrame(const OpenSim::PhysicalFrame& aFrame);
     /** Find this Station's location in any Frame */
-    SimTK::Vec3 findLocationInFrame(const SimTK::State& s, const OpenSim::Frame& aFrame) const;
+    SimTK::Vec3 findLocationInFrame(const SimTK::State& s,
+                                    const OpenSim::Frame& frame) const;
 private:
+    /** Calculate this Station's location in Ground */
+    SimTK::Vec3 calcLocationInGround(const SimTK::State& state) const override;
+    /** Calculate the velocity of this Station with respect to and expressed in
+        ground as a function of the state. */
+    SimTK::Vec3 calcVelocityInGround(const SimTK::State& state) const override;
+    /** Calculate the acceleration of this Station with respect to and 
+        expressed in ground as a function of the state. */
```

Comment:
IMO the 3 doxygen blocks above do not add anything. I think comments here are only necessary if they describe some quirk about this specific implementation of the virtual methods.